app/services/edge_cache/buster.rb

```diff
@@ -0,0 +1,115 @@
+module EdgeCache
+  class Buster
+    PROVIDERS = %w[fastly nginx].freeze
+
+    def initialize(path)
+      # TODO: (Vaidehi Joshi) - Right now, we are checking that nginx is
+      # available on every purge request/call to this buster service. If we are going
+      # to bust multiple paths, we should be able to check that nginx is
+      # available just once, and persist it on the class with @provider_available?.
+      # Then, we could allow for an array of @paths = [] to be passed in,
+      # and on single buster instance could bust multiple paths in order.
+
+      @path = path
+      @provider = set_provider
+    end
+
+    def self.call(*args)
+      new(*args).call
+    end
+
+    def call
+      return unless PROVIDERS.include?(provider)
+
+      case provider
```

Comment:
@citizen428 Great call here. I decided to go with the former option, just because I want to save the "split out into separate classes" work for a separate PR. I plan to make this service more efficient (see [this TODO comment](https://github.com/forem/forem/pull/10369#discussion_r490538857)), so I'll plan to split out the service into classes at that point 👍

Reply:
Ah, that makes sense, totally ok to do this in a follow-up PR (or not at all, as you see fit).